docs/examples/read_write_files.md

```diff
@@ -0,0 +1,140 @@
+# Read and Write Files
+
+Interacting with the filesystem to read and write files is a basic requirement
+of most development projects. Deno provides a number of ways to do this via the
+[standard library](https://deno.land/std) and the
+[Deno runtime API](https://doc.deno.land/builtin/stable).
+
+As highlighted in the [Fetch Data example](./fetch_data) Deno restricts access
+to Input / Output by default for security reasons. So when interacting with the
+filesystem the `--allow-read` and `--allow-write` flags must be used with the
+`deno run` command.
+
+## Read
+
+The Deno runtime API makes it possible to read text files via the
+`readTextFile()` method, it just requires a path string or URL object. The
+method returns a promise which provides access to the file's text data.
+
+**Command:** `deno run --allow-read read.ts`
+
+```js
+async function readFile(path: string): Promise<string> {
+  return await Deno.readTextFile(new URL(path, import.meta.url));
+}
+
+const text = readFile("./people.json");
+
+text.then((response) => console.log(response));
+
+/**
+ * Output:
+ *
+ * [
+ *   {"id": 1, "name": "John", "age": 23},
+ *   {"id": 2, "name": "Sandra", "age": 51},
+ *   {"id": 5, "name": "Devika", "age": 11}
+ * ]
+ */
+```
+
+The Deno standard library enables more advanced interaction with the filesystem
+and provides methods to read and parse files. The `readJson()` and
+`readJsonSync()` methods allow developers to read and parse files containing
+JSON. All these methods require is a valid file path string which can be
+generated using the `fromFileUrl()` method.
+
+In the example below the `readJsonSync()` method is used, for asynchronus
+execution use the `readJson()` method.
+
+Currently some of this functionality is marked as unstable so the `--unstable`
+flag is required along with the `deno run` command.
+
+**Command:** `deno run --unstable --allow-read read.ts`
+
+```js
+import { readJsonSync } from "https://deno.land/std@0.65.0/fs/mod.ts";
+import { fromFileUrl } from "https://deno.land/std@0.65.0/path/mod.ts";
+
+function readJson(path: string): object {
+  const file = fromFileUrl(new URL(path, import.meta.url));
+  return readJsonSync(file) as object;
+}
+
+console.log(readJson("./people.json"));
+
+/**
+ * Output:
+ *
+ * [
+ *   {"id": 1, "name": "John", "age": 23},
+ *   {"id": 2, "name": "Sandra", "age": 51},
+ *   {"id": 5, "name": "Devika", "age": 11}
+ * ]
+ */
+```
+
+## Write
+
+The Deno runtime API allows developers to write text to files via the
+`writeTextFile()` method. It just requires a file path and text string. On
+success the method will return an empty promise.
```

Comment:
> On success the method will return an empty promise.

The method returns a promise which resolves when the file was successfully written.